packages/strapi-design-system/src/Select/SingleSelect.tsx

```diff
@@ -0,0 +1,330 @@
+import { CarretDown, Cross } from '@strapi/icons';
+import styled from 'styled-components';
+
+import { Field, FieldError, FieldHint, FieldLabel } from '../Field';
+import { Stack } from '../Stack';
+import { Box } from '../Box';
+import { Flex } from '../Flex';
+import { Typography } from '../Typography';
+
+import { useId } from '../helpers/useId';
+
+import { useIntersection } from '../hooks/useIntersection';
+
+import { getThemeSize, inputFocusStyle } from '../themes/utils';
+
+export interface SingleSelectProps {
+  children: React.ReactNode;
+  /**
+   * @default "Clear"
+   */
+  clearLabel?: string;
+  /**
+   * @default (value) => value.toString()
+   */
+  customizeContent?: (value: string | number) => string;
+  disabled?: boolean;
+  error?: string | boolean;
+  hint?: string | React.ReactNode | React.ReactNode[];
+  id?: string | number;
+  label: string;
+  labelAction?: React.ReactElement;
+  onChange?: (value: string | number) => void;
+  onClear?: React.MouseEventHandler<HTMLDivElement>;
+  onReachEnd?: (entry: IntersectionObserverEntry) => void;
+  placeholder?: string;
+  required?: boolean;
+  /**
+   * @deprecated This prop is no longer required and will be removed in v2 of the DS.
+   * It has no effect on the component.
+   */
+  selectButtonTitle?: string;
+  /**
+   * @default "M"
+   */
+  size?: 'S' | 'M';
+  startIcon?: React.ReactElement;
+  value?: string | number;
+}
+
+export const SingleSelect = ({
+  children,
+  clearLabel = 'Clear',
+  customizeContent = (value) => value.toString(),
+  disabled,
+  error,
+  hint,
+  id,
+  label,
+  labelAction,
+  onChange,
+  onClear,
+  onReachEnd,
+  placeholder,
+  required,
+  selectButtonTitle,
+  startIcon,
+  size = 'M',
+  value: passedValue,
+}: SingleSelectProps) => {
+  /**
+   * Used for the intersection observer
+   */
+  const viewportRef = React.useRef<HTMLDivElement>(null);
+
+  const clearRef = React.useRef(null);
+
+  /**
+   * These values are drawn out from the internals of the Radix component
+   * We can then use them to react to visual changes for the component
+   */
+  const [internalValue, setInternalValue] = React.useState('');
+  const [internalIsOpen, setInternalIsOpen] = React.useState(false);
+
+  const generatedId = useId('select', id);
+
+  const hintId = `${generatedId}-hint`;
+  const errorId = `${generatedId}-error`;
+
+  const handleValueChange: Pick<RadixSelect.SelectProps, 'onValueChange'>['onValueChange'] = (value) => {
+    /**
+     * If it's being externally managed then we shouldn't
+     * both setting our copy of the internal value.
+     */
+    if (onChange) {
+      const shouldBeNumber = typeof passedValue === 'number';
+      onChange(shouldBeNumber ? Number(value) : value);
+    } else {
+      setInternalValue(value);
+    }
+  };
+
+  const handleClearClick: React.MouseEventHandler<HTMLDivElement> = (e) => {
+    console.log('calling clear');
+    if (onClear) {
+      onClear(e);
+    }
```

Comment:
You have a couple of these cases: should we just make them noop functions with a default assignment?

Reply:
Can do if you feel that strongly about it. I prefer this because it's a bit more explicit but idm - what do you wanna do?

Reply:
Ok, keep it then 👍🏼